milatools/cli/vscode_utils.py

```diff
@@ -36,3 +45,186 @@ def get_expected_vscode_settings_json_path() -> Path:
 
 def vscode_installed() -> bool:
     return bool(shutil.which(os.environ.get("MILATOOLS_CODE_COMMAND", "code")))
+
+
+EXTENSIONS_ARCHIVE_NAME = "vscode-extensions.tar.gz"
+EXTRACTED_VSCODE_EXTENSIONS_FILE = "extracted_vscode_extensions.txt"
+TRANSFERED_VSCODE_EXTENSIONS_FILE = "transferred_vscode_extensions.txt"
+
+
+def copy_vscode_extensions_to_remote(
```

Comment:
I'm wondering if a `rsync --update --partial` could be a simpler solution to handle all of the cases we have here?

Reply:
Listing out all the tiny files in each plugin directory takes forever on the $HOME filesystem of DRAC clusters. That's why I'm avoiding anything that lists out the contents of the directories.

Reply:
Ah I wasn't sure if `rsync` what listing or not. Even without the `--update` flag? Maybe we could make the check of which extensions needs update then use rsync only for some files?

A possibly much efficient tool on DRAC clusters could be globus but we would need to instantiate our own client on the cluster and for the moment that is not a a situation that is going to happen soon

Reply:
I think something like this would be very efficient : `https://stackoverflow.com/a/71384954`

`tar cjvf - /etc | ssh user@hostname "tar xvjf - --keep-newer-files"`

This would not ls files, only check if a file is newer or not

Reply:
Ah yes very true for the cross platform. I would tend to prefer reusing existing battle tested tools instead of reimplementing and adding code we have to maintain but if it makes the tool unusable on Windows that is a real issue